src/components/ListItem.jsx

```diff
@@ -1,22 +1,38 @@
 import { useEffect, useState } from 'react';
 import './ListItem.css';
 import { updateItem } from '../api';
-import { useStateWithStorage } from '../utils';
+import { handleNextPurchaseDate, ONE_DAY_IN_MILLISECONDS } from '../utils';
 import { increment } from 'firebase/firestore';
 
-export function ListItem({ name, itemId, purchaseTimestamp }) {
+// move current date outside of component to use it thoroughout the component
+const currentDate = new Date();
+
+export function ListItem({ item, listPath }) {
 	const [isPurchased, setIsPurchased] = useState(false);
-	const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
+	// ---------------------------------------------------------------
+	// the path from the hook is glitching in case the user has multiple empty lists:
+	// navigates to the first empty list instead of the path that was clicked on
+	// const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
+	// ---------------------------------------------------------------
+	const {
+		name,
+		id,
+		dateLastPurchased,
+		dateCreated,
+		dateNextPurchased,
+		totalPurchases,
+	} = item;
 
 	useEffect(() => {
-		if (!purchaseTimestamp) {
+		if (!dateLastPurchased) {
 			setIsPurchased(false);
 			return;
 		}
-		const purchaseDate = purchaseTimestamp.toDate();
-		const oneDayLater = new Date(purchaseDate.getTime() + 24 * 60 * 60 * 1000);
-		const currentDate = new Date();
-		if (purchaseTimestamp) {
+		const purchaseDate = dateLastPurchased.toDate();
+		const oneDayLater = new Date(
+			purchaseDate.getTime() + ONE_DAY_IN_MILLISECONDS,
```

Comment:
For this test error, you'll just have to add this const as part of the mocked module. LMK if you need help here

Reply:
Fixed, thank you! Passed all tests